starter_code/elevator.js

```diff
@@ -1,19 +1,66 @@
+/* jshint esversion:6 */
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
-  }
-
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.direction = "up";
+    this.waitingList = [];
+    this.passengers = [];
+    this.requests = [];
+  }
+
+  start() {
+    let intervalId = setInterval(() => {
+      this.update();
```

Comment:
No hubiera sido mala idea asignar el Id del intervalo como una propiedad del objeto Elevator. Algo así como "this.intervalId" y podríamos acceder dentro de cualquier función del objeto.